source/client-side-operations-timeout/tests/pending-response.yml

```diff
@@ -0,0 +1,415 @@
+description: "Connection churn is prevented by reading pending responses during connection checkout"
+schemaVersion: "1.26"
+runOnRequirements:
+  - minServerVersion: "4.4"
+    # TODO(SERVER-96344): When using failpoints, mongos returns MaxTimeMSExpired
+    # after maxTimeMS, whereas mongod returns it after
+    # max(blockTimeMS, maxTimeMS). Until this ticket is resolved, these tests
+    # will not pass on sharded clusters.
+    topologies: ["single", "replicaset"]
+createEntities:
+  - client:
+      id: &failPointClient failPointClient
+      useMultipleMongoses: false
+  - client:
+      id: &client client
+      uriOptions:
+        maxPoolSize: 1
+      useMultipleMongoses: false
+      observeEvents:
+        - commandFailedEvent
+        - commandSucceededEvent
+        - connectionCheckedOutEvent
+        - connectionCheckedInEvent
+        - connectionClosedEvent
+        - connectionPendingResponseSucceeded
+        - connectionPendingResponseStarted
+        - connectionPendingResponseFailed
+  - database:
+      id: &database test
+      client: *client
+      databaseName: *database
+  - collection:
+      id: &collection coll
+      database: *database
+      collectionName: *collection
+initialData:
+  - collectionName: *collection
+    databaseName: *database
+    documents: []
+tests:
+  # Attempting a pending response read on a non-timeout operation that can
+  # immediately read from the TCP buffer should complete the pending read and
+  # the connection should be checked out.
+  - description: "non-timeout op with response and no error"
+    operations:
+      # Run a ping command to pre-load the pool with a connection.
+      - name: runCommand
+        object: *database
+        arguments:
+          command:
+            ping: 1
+          commandName: ping
+      # Create a failpoint to block the first operation.
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: {times: 1}
+            data:
+              failCommands: ["insert"]
+              blockConnection: true
+              blockTimeMS: 100
+      # Execute operation with timeout less than block time.
+      - name: insertOne
+        object: *collection
+        arguments:
+          timeoutMS: 75
+          document: {_id: 3, x: 1}
+        expectError:
+          isTimeoutError: true
+      # Execute a subsequent operation to complete the read.
+      - name: findOne
```

Comment:
`findOne` operation is optional and not all drivers implements it (we do not support is in CSharp Driver for example). Can we replace it with `find`?